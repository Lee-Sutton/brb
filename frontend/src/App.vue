<template>
  <div id="app">
    <div id="nav">
      <Navbar></Navbar>
    </div>
    <SideNav></SideNav>
    <div class="container" style="margin-top: 5%">
      <router-view/>
    </div>
  </div>
</template>

<script>
  import Navbar from '@/components/Navbar.vue';
  import SideNav from '@/components/SideNav.vue';

  export default {
    components: {
      Navbar,
      SideNav,
    },
  }

</script>

