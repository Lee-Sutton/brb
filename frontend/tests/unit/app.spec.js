import {mount} from '@vue/test-utils';
import App from '@/App'

const mockRoute = {
  $route: {
    meta: {
      title: 'Dummy Title'
    }
  }
};

test('It should render', () => {
  const wrapper = mount(App, {
    stubs: ['router-link', 'router-view'],
    mocks: {
      ...mockRoute,
      $store: {
        state: {isLoggedIn: true}
      }
    }
  });
  expect(wrapper.html).toBeTruthy();
});

