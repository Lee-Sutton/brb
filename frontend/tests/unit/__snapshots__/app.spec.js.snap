// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`it should render 1`] = `
<div id="wrapper" class="d-flex toggle">
  <div id="sidebar-wrapper" class="bg-light border-right">
    <div class="sidebar-heading">Golf Buddy</div>
    <div class="list-group list-group-flush">
      <router-link-stub to="scores" class="list-group-item list-group-item-action bg-light">Scores</router-link-stub> <a href="#" class="list-group-item list-group-item-action bg-light">Drills</a> <a href="#" class="list-group-item list-group-item-action bg-light">Handicap</a> <a href="#" class="list-group-item list-group-item-action bg-light">Short Game</a> <a href="#" class="list-group-item list-group-item-action bg-light">Long Game</a> <a href="#" class="list-group-item list-group-item-action bg-light">Courses</a> <a href="#" class="list-group-item list-group-item-action bg-light">Find Friends</a>
    </div>
  </div>
  <div id="page-content-wrapper">
    <div>
      <nav class="navbar navbar navbar-expand-lg navbar-light bg-light navbar-light bg-info navbar-expand">
        <h2>Dummy Title</h2>
        <div id="nav-collapse" class="navbar-collapse collapse" style="display: none;">
          <ul class="navbar-nav ml-auto">
            <form class="form-inline">
              <li class="nav-item"><a target="_self" href="#" class="nav-link"><input type="text" placeholder="Find a course" class="mr-sm-2 form-control form-control-sm"></a></li>
            </form>
            <ul class="navbar-nav">
              <li class="nav-item"><a target="_self" href="scores" class="nav-link">Add Score</a></li>
              <li class="nav-item"><a target="_self" href="scores" class="nav-link">Handicap</a></li>
            </ul>
            <li class="nav-item b-nav-dropdown dropdown"><a aria-haspopup="true" aria-expanded="false" target="_self" href="#" class="nav-link dropdown-toggle"><em>User</em></a>
              <ul tabindex="-1" class="dropdown-menu dropdown-menu-right">
                <li><a role="menuitem" target="_self" href="#" class="dropdown-item">Profile</a></li>
                <li><a role="menuitem" target="_self" href="#" class="dropdown-item">Sign Out</a></li>
              </ul>
            </li>
          </ul>
        </div>
      </nav>
    </div>
    <div id="router-view" class="container">
      <router-view-stub></router-view-stub>
    </div>
  </div>
</div>
`;
